File: cases/taint/kotlin/call-context-positive/CallContextPositive.kt
package dataflowbench

object CallContextPositive {
    fun dfb_source(): Int { // DFB-SOURCE: call-context-input
        return 1
    }

    fun relay(value: Int): Int { // DFB-WITNESS: call-context-relay
        return value
    }

    fun dfb_sink(value: Int) {} // DFB-SINK: call-context-sink

    fun run() {
        val tainted = relay(dfb_source())
        val clean = relay(0)
        dfb_sink(tainted)
    }
}

File: cases/taint/go/call-context-positive/call_context_positive.go
package dataflowbench

func dfb_source() int { // DFB-SOURCE: call-context-input
	return 1
}

func relay(value int) int { // DFB-WITNESS: call-context-relay
	return value
}

func dfb_sink(value int) {} // DFB-SINK: call-context-sink

func run() {
	tainted := relay(dfb_source())
	_ = relay(0)
	dfb_sink(tainted)
}
